dev-util/boost-build/boost-build-1.70.0.ebuild

```diff
@@ -0,0 +1,143 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 )
+inherit flag-o-matic python-single-r1 toolchain-funcs
+
+MY_PV="$(ver_rs 1- _)"
+
+DESCRIPTION="A system for large project software construction, simple to use and powerful"
+HOMEPAGE="https://boostorg.github.io/build/"
+SRC_URI="https://downloads.sourceforge.net/project/boost/boost/${PV}/boost_${MY_PV}.tar.bz2"
+
+LICENSE="Boost-1.0"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~ppc-aix ~x64-cygwin ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+IUSE="examples python test"
+RESTRICT="test"
+
+RDEPEND="python? ( ${PYTHON_DEPS} )
+	!<dev-libs/boost-1.35.0
+	!<=dev-util/boost-build-1.35.0-r1"
+DEPEND="${RDEPEND}
+	test? (
+		sys-apps/diffutils
+		${PYTHON_DEPS}
+	)"
+
+REQUIRED_USE="
+	python? ( ${PYTHON_REQUIRED_USE} )
+	test? ( ${PYTHON_REQUIRED_USE} )"
+
+S="${WORKDIR}/boost_${MY_PV}/tools/build/src"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-1.48.0-disable_python_rpath.patch"
+	"${FILESDIR}/${PN}-1.50.0-respect-c_ld-flags.patch"
+	"${FILESDIR}/${PN}-1.49.0-darwin-gentoo-toolchain.patch"
+	"${FILESDIR}/${PN}-1.52.0-darwin-no-python-framework.patch"
+	"${FILESDIR}/${PN}-1.54.0-support_dots_in_python-buildid.patch"
+	"${FILESDIR}/${PN}-1.55.0-ppc-aix.patch"
+	"${FILESDIR}/${PN}-1.62.0-sparc-no-default-flags.patch"
+	"${FILESDIR}/${PN}-1.66.0-add-none-feature-options.patch"
+)
+
+pkg_setup() {
+	if use python || use test; then
+		python-single-r1_pkg_setup
+	fi
+}
+
+src_unpack() {
+	tar xojf "${DISTDIR}/${A}" boost_${MY_PV}/tools/build || die "unpacking tar failed"
+}
+
+src_prepare() {
+	default
+
+	pushd ../ >/dev/null || die
+	eapply "${FILESDIR}/${PN}-1.54.0-fix-test.patch"
+	popd >/dev/null || die
+
+	# Remove stripping option
+	# Fix python components build on multilib systems, bug #496446
+	cd "${S}/engine" || die
+	sed -i \
+		-e 's|-s\b||' \
+		-e "/libpython/s/lib ]/$(get_libdir) ]/" \
+		build.jam || die "sed failed"
+
+	# Force regeneration
+	rm jambase.c || die
+
+	# This patch allows us to fully control optimization
+	# and stripping flags when bjam is used as build-system
+	# We simply extend the optimization and debug-symbols feature
+	# with empty dummies called 'none'
+	cd "${S}" || die
+	sed -i \
+		-e 's/\(off speed space\)/\1 none/' \
+		-e 's/\(debug-symbols      : on off\)/\1 none/' \
```

Comment:
Wouldn't it be better not to rely on exact whitespace? And ofc again, patch would be better than relying on sed not stopping to match here.